README.md

```diff
@@ -226,6 +229,51 @@ The only required key/value in the signing_info dictionary is 'identity'.
 See the **SIGNED PACKAGES** section of the man page for `pkgbuild` or the **SIGNED PRODUCT ARCHIVES** section of the man page for `productbuild` for more information on the signing options.
 
 
+### Package notarization
+
+You may notarize **SIGNED PACKAGES** as part of the build process by adding a notarization dictionary to the build\_info.plist:
+
+```plist
+    <key>notarization</key>
+    <dict>
+        <key>username</key>
+        <string>john.appleseed@apple.com</string>
+        <key>password</key>
+        <string>@keychain:AC_PASSWORD</string>
+        <key>stapler_timeout</key>
+        <integer>120</integer>
+    </dict>
+```
+
+or, in JSON format in a build-info.json file:
+
+```json
+    "notarization": {
+        "username": "john.appleseed@apple.com",
+        "password": "@keychain:AC_PASSWORD",
+        "stapler_timeout":  120
+    }
+```
+
+The only required keys/values in the notarization dictionary are `username` and `password`:
+
+- `username` is the login email adress of your developer Apple ID.
+- `password` is the 2FA app specific password. For information about the password and saving
+it to the login keychain see the web page [Customizing the Notarization Workflow](https://developer.apple.com/documentation/security/notarizing_your_app_before_distribution/customizing_the_notarization_workflow).
+
+
+Notarization process is described on Apple web page [Customizing the Notarization Workflow](https://developer.apple.com/documentation/security/notarizing_your_app_before_distribution/customizing_the_notarization_workflow).
+
+`munki-pkg` basically runs two commands:
+
+```shell
+xcrun altool --notarize-app --primary-bundle-id "com.github.munki.pkg.munki-kickstart" --username "john.appleseed@apple.com" --password "@keychain:AC_PASSWORD" --file munki_kickstart.pkg
+xcrun stapler staple munki_kickstart.pkg
+```
+
+There is a time delay between successfull upload of a signed package to the notary service using `altool` and availability of the staple from the notary service `stapler` can use.
```

Comment:
s/successfull/successful/

Reply:
I'm not sure I understand how `stapler` knows what to do -- if I have multiple notarization requests, how does it know which one to apply to the thing I'm trying to staple?

Reply:
Every package has unique code signature, correct?
There should is the public key so everyone can verify the package signature. Public key is signed by Apple so we can verify it's validity.

Speculation: Should not be hard to find record in Apple notary services database based on some information from package signature.

If you try to upload the exact same package multiple times Apple notary service is not happy about that. However if you upload "same" package to notary service after each munki-pkg build there should be at least a little difference  in signature (timestamp?) -> not the same package -> no errors.